numpy/core/tests/test_dtype.py

```diff
@@ -794,6 +794,36 @@ class PaddedStruct(ctypes.Structure):
         ], align=True)
         self.check(PaddedStruct, expected)
 
+    def test_bit_fields(self):
+        class BitfieldStruct(ctypes.Structure):
+            _fields_ = [
+                ('a', ctypes.c_uint8, 7),
+                ('b', ctypes.c_uint8, 1)
+            ]
```

Comment:
Previously this produced `dtype([('a', 'u1', (7,)), ('b', 'u1')], align=True)`, which was garbage.